rpcs3/Emu/RSX/overlay_controls.h

```diff
@@ -119,33 +120,58 @@ namespace rsx
 			{
 				//Init glyph
 				std::vector<u8> bytes;
+				std::vector<std::string> font_dirs;
 				std::vector<std::string> fallback_fonts;
 #ifdef _WIN32
-				std::string font_dir = "C:/Windows/Fonts/";
+				font_dirs.push_back("C:/Windows/Fonts/");
 				fallback_fonts.push_back("C:/Windows/Fonts/Arial.ttf");
 #else
 				char *home = getenv("HOME");
 				if (home == nullptr)
 					home = getpwuid(getuid())->pw_dir;
 
-				std::string font_dir = home;
-				if (home[font_dir.length() - 1] == '/')
-					font_dir += ".fonts/";
+				font_dirs.push_back(home);
+				if (home[font_dirs[0].length() - 1] == '/')
+					font_dirs[0] += ".fonts/";
 				else
-					font_dir += "/.fonts/";
+					font_dirs[0] += "/.fonts/";
 
 				fallback_fonts.push_back("/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf"); //ubuntu
 				fallback_fonts.push_back("/usr/share/fonts/TTF/DejaVuSans.ttf"); //arch
 #endif
-				//Also attempt to load from dev_flash as a last resort
+				//Search dev_flash for the font too
+				font_dirs.push_back(Emu.GetEmuDir() + "dev_flash/data/font/");
+				font_dirs.push_back(Emu.GetEmuDir() + "dev_flash/data/font/SONY-CC/");
+
+				//Attempt to load a font from dev_flash as a last resort
 				fallback_fonts.push_back(Emu.GetEmuDir() + "dev_flash/data/font/SCE-PS3-VR-R-LATIN.TTF");
 
-				std::string requested_file = font_dir + ttf_name + ".ttf";
-				std::string file_path = requested_file;
+				//Attemt to load requested font
+				std::string file_path;
+				bool font_found = false;
+				for (auto& font_dir : font_dirs)
+				{
+					std::string requested_file = font_dir + ttf_name;
```

Comment:
If its necessary to append .ttf anyway there is no need for the option to supply the font name with the ttf. The caller should make sure of that otherwise this can hide bad code.

Reply:
The reason for supporting both with and without an extension was to support cases where the extensions is capitalized (`.TTF`) as is the case with fonts in the firmware.

I didn't want to touch existing code to force the caller to include the extension _always_. Is this what you're suggesting I do?

Reply:
That will work only if you do `element.set_font("FULL_NAME.TTF", size)`. Thats strange to include the extension when loading the font by name but I'll allow it, mostly because I do not want to actually rewrite the font search code to properly handle corner cases lol. Really should use directory traversal on linux in case the file name does not match exactly. I already had this problem on my system vs an arch user getting different results because of it.

Reply:
Oh... I didn't know it was already possible to set the name exactly.

>but I'll allow it

👍

Reply:
I mean with your solution it will become possible to pass full name. My method always appends .ttf in lower case which will break exact font names. So you can currently set full name but it will fail as you get double extension.

Reply:
Yup, that's why I did it.